pkg/runtime/local/local.go

```diff
@@ -106,47 +95,31 @@ func (r *Runtime) RunGadget(gadgetCtx runtime.GadgetContext) (out []byte, err er
 		setter.SetEventEnricher(operatorInstances.Enrich)
 	}
 
+	log.Debug("calling operator.PreGadgetRun()")
 	err = operatorInstances.PreGadgetRun()
 	if err != nil {
 		return nil, fmt.Errorf("gadget prerun: %w", err)
 	}
-	defer operatorInstances.PostGadgetRun()
+	defer func() {
+		log.Debug("calling operator.PostGadgetRun()")
+		operatorInstances.PostGadgetRun()
+	}()
 
-	if startstop, ok := gadgetInstance.(gadgets.StartStopAltGadget); ok {
-		log.Debugf("calling gadget.StartAlt()")
-		err := startstop.StartAlt()
-		if err != nil {
-			return out, fmt.Errorf("starting gadget: %w", err)
-		}
-		defer func() {
-			log.Debugf("calling gadget.StopAlt()")
-			startstop.StopAlt()
-		}()
-	} else if startstop, ok := gadgetInstance.(gadgets.StartStopGadget); ok {
-		log.Debugf("calling gadget.Start()")
-		err := startstop.Start()
-		if err != nil {
-			return out, fmt.Errorf("starting gadget: %w", err)
-		}
-		defer func() {
-			log.Debugf("calling gadget.Stop()")
-			startstop.Stop()
-		}()
-	} else if run, ok := gadgetInstance.(gadgets.RunGadget); ok {
+	if run, ok := gadgetInstance.(gadgets.RunGadget); ok {
 		log.Debugf("calling gadget.Run()")
-		err := run.Run()
+		err := run.Run(gadgetCtx)
 		if err != nil {
 			return out, fmt.Errorf("running gadget: %w", err)
 		}
+	} else if runWithResult, ok := gadgetInstance.(gadgets.RunWithResultGadget); ok {
+		log.Debugf("calling gadget.RunWithResult()")
+		out, err = runWithResult.RunWithResult(gadgetCtx)
+		if err != nil {
+			return out, fmt.Errorf("running (with result) gadget: %w", err)
+		}
+	} else {
+		return nil, errors.New("gadget not runnable")
```

Comment:
nit: There is a mix of  `return out,...`  and `return nil, ...`. Do we really need to make the return values "named" now?

Reply:
Yep, "named" return is not needed anymore. I removed it.